src/Jobs/CreatePostgresUserForTenant.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\DB;
+use Stancl\Tenancy\Database\Contracts\TenantWithDatabase;
+
+class CreatePostgresUserForTenant implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(
+        protected TenantWithDatabase&Model $tenant,
+    ) {
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $name = $this->tenant->database()->getUsername() ?? $this->tenant->getTenantKey();
+        $password = $this->tenant->database()->getPassword() ?? 'password';
```

Comment:
Is `password` the default password if there's no password available in `getPassword()`?

Maybe a blank password would be more standard, that's the case with MySQL, not sure about pgsql.

Reply:
I see, the default should be password `postgres`

Reply:
Changed this entirely – since we can't use empty string passwords with Postgres, I added a static property to the RLS bootstrapper

```php
/** Mustn't return an empty string */
public static Closure|string $defaultPassword = 'password';
```

This is used both for the Postgres connection password config, and the newly created user.

The bootstrapper now has a public static method so that it can be used conveniently in both the bootstrapper and the user creation job.

```php
public static function getDefaultPassword(): string
{
    return is_string(static::$defaultPassword) ? static::$defaultPassword : (static::$defaultPassword)();
}
```

Reply:
Improved this so that the default password is configurable https://github.com/archtechx/tenancy/pull/1105/commits/0fa52e502027743d9181d03a5031f88ac0dc6476

Reply:
Mention this in the readme, so that people make sure to either have users that have passwords specified, or use a secure default password.

Alternatively, since this feature isn't really about separating users for security reasons, but rather for *convenience* (automatic scoping), wouldn't it make sense to use the DB password that's being used by the central DB?

Reply:
> since this feature isn't really about separating users for security reasons, but rather for convenience (automatic scoping), wouldn't it make sense to use the DB password that's being used by the central DB

I think this makes sense. I'll update the code

Reply:
> Mention this in the pr desc, so that people make sure to either have users that have passwords specified, or use a secure default password.

Is this done as well? Just mention that no password defaults to the one used for the central connection.